src/main.js

```diff
@@ -420,6 +421,14 @@ var manifestor = function(options) {
       if (viewerState().perspective === 'detail') {
         applyConstraints(_constraintBounds);
       }
+      // Open the main tile source when we reach the specified zoom level on it
+      if(event.zoom >= _transitionZoomLevel && event.refPoint) {
```

Comment:
Instead of using the refPoint of the zoom (which won't exist if the user hits the zoom button), just grab the center of the viewport (`viewer.viewport.getBounds().getCenter()`) and use that.